presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/TransformSingleValueInToEquality.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.iterative.rule;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.sql.planner.PlanNodeIdAllocator;
+import com.facebook.presto.sql.planner.SymbolAllocator;
+import com.facebook.presto.sql.planner.iterative.Lookup;
+import com.facebook.presto.sql.planner.iterative.Rule;
+import com.facebook.presto.sql.planner.plan.FilterNode;
+import com.facebook.presto.sql.planner.plan.PlanNode;
+import com.facebook.presto.sql.tree.ComparisonExpression;
+import com.facebook.presto.sql.tree.ComparisonExpressionType;
+import com.facebook.presto.sql.tree.Expression;
+import com.facebook.presto.sql.tree.InListExpression;
+import com.facebook.presto.sql.tree.InPredicate;
+import com.facebook.presto.sql.tree.Literal;
+
+import java.util.List;
+import java.util.Optional;
+
+public class TransformSingleValueInToEquality
+        implements Rule
+{
+    @Override
+    public Optional<PlanNode> apply(PlanNode node, Lookup lookup, PlanNodeIdAllocator idAllocator, SymbolAllocator symbolAllocator, Session session)
+    {
+        if (!(node instanceof FilterNode)) {
+            return Optional.empty();
+        }
+
+        FilterNode filter = (FilterNode) node;
+        if (!(filter.getPredicate() instanceof InPredicate)) {
+            return Optional.empty();
+        }
+
+        InPredicate inPredicate = (InPredicate) filter.getPredicate();
+        if (!(inPredicate.getValueList() instanceof InListExpression)) {
+            return Optional.empty();
+        }
+
+        // if the value list contains only one value expression and that value expression is a literal, convert the IN predicate to a equality predicate
+        List<Expression> inListExpressions = ((InListExpression) inPredicate.getValueList()).getValues();
+        if (inListExpressions.size() == 1 && inListExpressions.get(0) instanceof Literal) {
+            Literal literalExpression = (Literal) inListExpressions.get(0);
+            ComparisonExpression comparisonExpression = (inPredicate.getLocation().isPresent())
+                    ? new ComparisonExpression(inPredicate.getLocation().get(), ComparisonExpressionType.EQUAL, inPredicate.getValue(), literalExpression)
```

Comment:
Even if you preserve location, what use it will have? Location is used for err reporting to the user and err like "you have wrong `=` at .." when you have `in` there is not going to be helpful.